junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java

```diff
@@ -141,4 +140,59 @@ public static void assertPackageNameIsValid(String packageName) {
 		}
 	}
 
+	/**
+	 * Collection of utilities for working with qualified names in Java.
+	 *
+	 * <h3>DISCLAIMER</h3>
+	 *
+	 * <p>These utilities are intended solely for usage within the JUnit framework
+	 * itself. <strong>Any usage by external parties is not supported.</strong>
+	 * Use at your own risk!
+	 *
+	 * @since 1.5
+	 */
+	@API(status = INTERNAL, since = "1.5")
+	static class JavaNameUtils {
+
+		private static final List<String> RESTRICTED_KEYWORDS = Arrays.asList("strictfp", "assert", "enum", "_",
```

Comment:
:o: I think this should be a unmodifiable `Set` which also should improve performance of `contains()`.

Reply:
+1

Reply:
I don't understand why I am not able to use Java 9's `Set.of()` syntax here?

Reply:
Because we use the '--release 8' compiler flag for our main/production code. This forbids usage of Java 9+ API here.
Test sources don't use the release flag and may use Java 12 syntax and API. This might shift to Java 13, soon.

Reply:
I see!